scipy/sparse/sparsetools/csr.h

```diff
@@ -745,7 +745,7 @@ void csr_binop_csr_general(const I n_row, const I n_col,
         for(I jj = 0; jj < length; jj++){
             T result = op(A_row[head], B_row[head]);
 
-            if(result != 0){
+            if(result != 0 || A_row[head] == 0){
```

Comment:
Why only check `A_row[head]` here, and not `B_row[head]`?

Reply:
As far as I could understand its working only those indices are checked for which there is a non-zero value (or explicitly defined zero) in at least one matrix so if result is zero and one matrix has zero value then other should also have zero value at that index (and one of the matrix must have a explicit zero at that index).